test/hotspot/jtreg/serviceability/sa/ClhsdbTestAllocationMerge.java

```diff
@@ -0,0 +1,68 @@
+/*
+ * Copyright (c) 2024, Oracle and/or its affiliates. All rights reserved.
+ * DO NOT ALTER OR REMOVE COPYRIGHT NOTICES OR THIS FILE HEADER.
+ *
+ * This code is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License version 2 only, as
+ * published by the Free Software Foundation.
+ *
+ * This code is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+ * version 2 for more details (a copy is included in the LICENSE file that
+ * accompanied this code).
+ *
+ * You should have received a copy of the GNU General Public License version
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ */
+
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import jdk.test.lib.apps.LingeredApp;
+import jtreg.SkippedException;
+
+/**
+ * @test
+ * @bug 8318682
+ * @summary Test clhsdb 'testdebuginfodecode' command
+ * @requires vm.hasSA
+ * @library /test/lib
+ * @run main/othervm -XX:CompileThresholdScaling=0.01 -XX:CompileCommand=compileonly,compiler.c2.HeapDumper::testIt -XX:CompileCommand=exclude,compiler.c2.HeapDumper::dummy ClhsdbTestAllocationMerge
```

Comment:
`LingeredAppWithAllocationMerge` instead of `compiler.c2.HeapDumper`

Reply:
Those options are actually in the wrong place since they actually need to be passed to the lingered app but it also seems like they are wrong.  I'd confirmed that the the test as written actually failed without the SA changes but when I added these arguments to the launch of the lingered app it no longer failed.  So since it properly detects the problem without any arguments I'm just going to drop them from the @run directive.